src/Microsoft.PowerShell.Commands.Management/commands/management/Navigation.cs

```diff
@@ -2699,7 +2699,7 @@ protected override void ProcessRecord()
                         try
                         {
                             System.IO.DirectoryInfo di = new(providerPath);
-                            if (di != null && (di.Attributes & System.IO.FileAttributes.ReparsePoint) != 0)
+                            if (di != null && di.Attributes.HasFlag(System.IO.FileAttributes.ReparsePoint) && (di.Attributes.HasFlag(System.IO.FileAttributes.Directory) && !shouldRecurse))
```

Comment:
I wonder why do we need the code at all if now FileSystemProvider correctly process all cases? (We can remove the code block from line 2696 to line 2713.)

Reply:
The code was added to make `remove-item -literalpath *` work for Registry provider.  I'll add to the comment.

Reply:
Registry provider? In line 2697 we check FileSystemProvider.

Reply:
The code is relying on DirectoryInfo class to determine if the path is a directory and this would only work with FileSystemProvider so this code is skipped for other providers (like Registry Provider where it would fail).  The way to think about this is if the path is FileSystemProvider path and a Directory, we do additional checks like reparse point to determine if the item should be treated as a file or not.  For other providers, these API calls would fail so they are skipped.